### ISSUE TITLE
Autocomplete width resize when active/focus

### ISSUE THREAD

**Issue by rigorcadiz** [akveo/react-native-ui-kitten#1736]
## 🐛 Bug Report

After updating from Expo 46 to Expo 48. The Autocomplete component no longer behaves at it is before. I checked the ui-kitten dependencies, and still has the same version, so probably due to update of react-native? 
from `0.69.9` to `0.71.71`

![Screenshot_2023-05-06-00-32-51-57_90e51a8aee41f7443aa19cc8cba36488](https://user-images.githubusercontent.com/102206810/236516425-217dee05-e5fa-4f55-ae1c-9f184e94c8a0.jpg)

## UI Kitten and Eva version

| Package      | Version |
| ----------- | ----------- |
| @eva-design/eva      |  2.2.0      |
| @ui-kitten/components   |   5.3.1      |

## Environment information

```bash

```

**Comment by bataevvlad** [comment]
@rigorcadiz

Hello there!

Please provide code example for more clear understanding and reproducing your problem.

Sincerely, UI Kitten team

**Comment by rigorcadiz** [comment]
@bataevvlad 

Here is the sample code. I created a basic app to debug this using the following command:

`npx create-expo-app MyApp`

Here is the App.json

```
import { StatusBar } from 'expo-status-bar';
import { StyleSheet, View } from 'react-native';
import * as eva from '@eva-design/eva';
import { ApplicationProvider, Layout, Text, Autocomplete, AutocompleteItem } from '@ui-kitten/components';

export default function App() {
  return (
    <ApplicationProvider {...eva} theme={eva.light}>
      <View style={styles.container}>
        <Text>Open up App.js to start working on your app!</Text>
        <StatusBar style="auto" />
        <View style={{
          width: 300, height: 300, borderWidth: 1, padding: 10
        }}>
          <Autocomplete
            placeholder='Place your Text'
          >
            <AutocompleteItem title='Option 1'/>
            <AutocompleteItem title='Option 2'/>
            <AutocompleteItem title='Option 3'/>
          </Autocomplete>
        </View>
      </View>
    </ApplicationProvider>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
```
This is my package.json:
```
{
  "name": "myapp",
  "version": "1.0.0",
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web"
  },
  "dependencies": {
    "@eva-design/eva": "^2.2.0",
    "@ui-kitten/components": "^5.3.1",
    "expo": "~48.0.15",
    "expo-status-bar": "~1.4.4",
    "react": "^18.0.0",
    "react-native": "0.71.7",
    "react-native-svg": "^13.9.0"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0"
  },
  "private": true
}
```

And I've got the same behavior with the screenshot above.

**Comment by rigorcadiz** [comment]
It used to be working normally before the update.

**Comment by TPK-MAXEDV** [comment]
We have the same problem. In our case, the problem seems to have something to do with the width property. 
If I set a fixed width to the autocomplete itself, it does all work. Percentage Values don't....

**Comment by rigorcadiz** [comment]
> We have the same problem. In our case, the problem seems to have something to do with the width property. If I set a fixed width to the autocomplete itself, it does all work. Percentage Values don't....

Yeah it works with fix width but there is also this other odd behavior that the component blinks/flash when you focus/activate it.. Does it happen on your end too or no?

**Comment by bataevvlad** [comment]
ok thanks @rigorcadiz we will check it out and come back soon!

**Comment by rigorcadiz** [comment]
Do we have an update here?

**Comment by bataevvlad** [comment]
@rigorcadiz I can advise you not to switch to a version above 0.71 react-native, the problem is associated with an update modal component of react-native, we are currently thinking about a solution, the fastest way would be to roll back to the old version 0.69 where you worked, I can not yet say how long it will take a fix

**Comment by rigorcadiz** [comment]
> @rigorcadiz I can advise you not to switch to a version above 0.71 react-native, the problem is associated with an update modal component of react-native, we are currently thinking about a solution, the fastest way would be to roll back to the old version 0.69 where you worked, I can not yet say how long it will take a fix

Too bad Apple no longer receive the SDK version on that react-native version that is why I was forced to update.

**Comment by ricardodolnl** [comment]
Any updates?

**Comment by ricardodolnl** [comment]
Can someone give me a status update on this bug? Any time indication for the fix? If it takes to long to fix I'll need to find another solution.

**Comment by Memz98** [comment]
@bataevvlad Has this been resolved? 

I'm using:
```
"@eva-design/eva": "^2.2.0",
"@ui-kitten/components": "^5.3.1",
```

and the issue still exists.

**Comment by johnstonbl01** [comment]
Also ran into this issue and just general inconsistencies when styling the list of autocomplete items. I ended up creating a custom autocomplete component for the one instance where I needed it, and figured I'd share it back here in case it helps.

```jsx
import { useState, useRef } from 'react';
import { Input, List, ListItem } from '@ui-kitten/components';
import { View, StylesSheet } from 'react-native';

const colors = ['red', 'blue', 'yellow', 'green'];
const filterColorsByName = (color, query) => color.toLowerCase().includes(query.toLowerCase());

export const CustomPopover = () => {
  const theme = useTheme();
  const ref = useRef();

  const [colorList, setColorList] = useState(colors);
  const [colorName, setColorName] = useState('');
  const [popoverVisible, setPopoverVisible] = useState(false);
  const [popoverPositions, setPopoverPositions] = useState(null);

  const onOptionSelect = useCallback(
    (color) => () => {
      setColorName(color);
      setPopoverVisible(false);
    },
    [colorList]
  );

  const onChangeText = (query) => {
    if (!popoverVisible && query.trim().length > 0) {
      setPopoverVisible(true);
    }

    setColorName(query);
    setColorList(colors.filter((color) => filterColorsByName(color, query)));
  };

  return (
    <Layout>
      <View
        ref={ref}
        onLayout={() => {
          ref?.current?.measureInWindow((x, y, width, height) => {
            setPopoverPositions({ x, y, width, height });
          });
        }}
      >
        <Input label="Color" value={colorName} onChangeText={onChangeText} />
      </View>
      <List
        style={[
          styles.autocompleteList,
          {
            display: popoverVisible ? 'flex' : 'none',
            borderColor: theme['color-basic-default-border'],
            top: popoverPositions?.y,
            left: popoverPositions?.x,
            width: popoverPositions?.width
          }
        ]}
        data={colorList}
        renderItem={({ item }) => <ListItem title={item} onPress={onOptionSelect(item)} />}
      />
    </Layout>
  );
};

const styles = StyleSheet.create({
  autocompleteList: {
    paddingVertical: 8,
    backgroundColor: 'white',
    borderWidth: 1,
    borderRadius: 4,
    position: 'absolute',
    zIndex: 1000
  }
});
```

It's not quite as fancy as the UI-Kitten one without the anchor, but it allowed me to move forward with similar UX. Hope this helps someone!

**Comment by glisom** [comment]
Thanks for sharing @johnstonbl01!

**Comment by RahulR100** [comment]
Seeing as this is still open... A simple fix for me was:

1. Wrap the `Autocomplete` element in a `<View>`
2. Give the `View` a `ref`
3. Add `style={{ width: ref.current?.offsetWidth }} || 100` to the `AutocompleteItem`

**Comment by rcarvalhocdmx** [comment]
Another solution that worked for me:

There is another solution that worked for me width mi using the "onLayout" prop o the View component and keeping the width in a state:

const Component = () => {
    const [ width, setWidth] = useState(400)
    const handleWidth = (event) => {
        const { width } = event.nativeEvent.layout
        setWidth(width)
    }
    return (
          <Layout>
                  <View onLayout={handleWidth}>
                      <Autocomplete
                           style={{ width }}
                         .....>
                         ....
                    </Autocomplete>
          </Layout/>
    )
}

![Screenshot from 2024-01-26 00-10-47](https://github.com/akveo/react-native-ui-kitten/assets/141180428/14b99dc5-46a4-48ba-bc84-97ff868abc4d)

**Comment by sizuf** [comment]
Why is this abandoned? To me it seems significant enough to be fixed already...

**Comment by bataevvlad** [comment]
The Autocomplete component uses a dual-input approach (anchor input + popover input) which can cause a brief visual transition when opening. The fullWidth property ensures the popover matches the anchor width, but slight visual differences may occur during the switch.

This is tracked as an improvement item for the upcoming release. Stay tuned!
